wgpu-core/src/device/mod.rs

```diff
@@ -1104,10 +1104,22 @@ impl<A: HalApi> Device<A> {
                         error,
                     })?;
             }
-            if writable_storage == WritableStorage::Yes
-                && entry.visibility.contains(wgt::ShaderStages::VERTEX)
-            {
-                required_features |= wgt::Features::VERTEX_WRITABLE_STORAGE;
+            if entry.visibility.contains(wgt::ShaderStages::VERTEX) {
+                if writable_storage == WritableStorage::Yes {
+                    required_features |= wgt::Features::VERTEX_WRITABLE_STORAGE;
+                }
+                if let Bt::Buffer {
+                    ty: wgt::BufferBindingType::Storage { .. },
+                    ..
+                } = entry.ty
+                {
+                    self.require_downlevel_flags(wgt::DownlevelFlags::VERTEX_STORAGE)
```

Comment:
Could we do something similar as the required_features, enumerate them then require them once at the bottom.

Reply:
Done as requested!